lib/qesapdeployment.pm

```diff
@@ -2029,4 +2031,83 @@ sub qesap_az_create_sas_token {
     record_info('GENERATE-SAS', $cmd);
     return script_output($cmd);
 }
+
+=head2 qesap_terraform_clean_up_retry
+
+    qesap_terraform_clean_up_retry()
+
+    Do terraform cleanup only when ansible failed on 'Timed out waiting for last boot time check'
+
+=cut
+
+sub qesap_terraform_clean_up_retry {
+    my $command = 'terraform';
```

Comment:
As there's only one command maybe we do not need a variable. maybe we can directly call
```
record_info('Cleanup', "Executing terraform cleanup");
    my @clean_up_cmd_rc = qesap_execute(verbose => '--verbose', cmd => 'terraform', cmd_options => '-d', timeout => 1200);
    if ($clean_up_cmd_rc[0] == 0) {
        diag("Terraform cleanup attempt #  PASSED.");
        record_info('Clean terraform', 'Terraform cleanup PASSED.');
    }
    else {
        diag('Terraform cleanup attempt #  FAILED.');
        record_info('Cleanup terraform', 'Terraform cleanup FAILED", result => 'fail');
    }
```

Reply:
Thanks, I would like to keep it. Maybe we will add more cleanup commands (such as doing ansible cleanup for "Missing sudo passwd" error).